pykoopman/observables/_radial_basis_functions.py

```diff
@@ -0,0 +1,317 @@
+        if polyharmonic_coeff < 0:
+            raise ValueError("polyharmonic_coeff must be a nonnegative float")
+        if rbf_type not in [
+            "thinplate",
+            "gauss",
+            "invquad",
+            "invmultquad",
+            "polyharmonic",
+        ]:
+            raise ValueError("rbf_type not of available type")
+        if type(include_states) != bool:
+            raise TypeError("include_states must be a boolean")
+        if centers is not None:
+            if int(n_centers) not in centers.shape():
+                raise ValueError(
+                    "n_centers is not equal to centers.shape[1]. "
+                    "centers must be of shape (n_input_features, "
+                    "n_centers). "
+                )
+        self.rbf_type = rbf_type
+        self.n_centers = int(n_centers)
+        self.centers = centers
+        self.kernel_width = kernel_width
+        self.polyharmonic_coeff = polyharmonic_coeff
+        self.include_states = include_states
+
+    def fit(self, x, y=None):
+        """
+        Fit to measurement data.
+
+        Parameters
+        ----------
+        x: array-like, shape (n_samples, n_input_features)
+            Measurement data to be fit.
+
+        y: None
+            Dummy parameter retained for sklearn compatibility.
+
+        Returns
+        -------
+        self: returns a fit :class:`pykoopman.observables.RadialBasisFunction` instance
+        """
+        n_samples, n_features = validate_input(x).shape
+
+        self.n_samples_ = n_samples
+        self.n_input_features_ = n_features
+        if self.include_states is True:
+            self.n_output_features_ = n_features * 1 + self.n_centers
+        elif self.include_states is False:
+            self.n_output_features_ = self.n_centers
+
+        x = validate_input(x)
+
+        if x.shape[1] != self.n_input_features_:
+            raise ValueError(
+                "Wrong number of input features. "
+                f"Expected x.shape[1] = {self.n_input_features_}; "
+                f"instead x.shape[1] = {x.shape[1]}."
+            )
+
+        if self.centers is None:
+            # Uniformly distributed centers
+            self.centers = random.rand(self.n_input_features_, self.n_centers)
+            # Change range to range of input features' range
+            for feat in range(self.n_input_features_):
+                xminmax = self._minmax(x[:, feat])
+
+                # Map to range [0,1]
+                self.centers[feat, :] = (
+                    self.centers[feat, :] - min(self.centers[feat, :])
+                ) / (max(self.centers[feat, :]) - min(self.centers[feat, :]))
+                # Scale to input features' range
+                self.centers[feat, :] = (
+                    self.centers[feat, :] * (xminmax[1] - xminmax[0]) + xminmax[0]
+                )
+
+        xlift = self._rbf_lifting(x)
+        self.measurement_matrix_ = x.T @ np.linalg.pinv(xlift.T)
```

Comment:
I see you use fitting to get the measurement_matrix, aka inverse mapping. 

Because for EDMD in this package, observables are computed through linear, polynomial, time delay assumes first N features are the system state. So I just use first N observables as return of the measurement/system state. There is no such feature of excluding system state, except this RBF basis that you developed. 

So to resolve this issue, I will revise the 'inverse' in existing observables using analytically filled self.measurement_matrix_.

Reply:
Please wait with the revision of inverse until I have finished my code changes. Note that I am not sure if you can easily change it, since it's a required method in the class the observables inherit from. More generally, yes, often the basis functions/observables include the state space directly, but not always. In the latter case, it is required to estimate the inverse mapping, e.g. using least squares.

Reply:
Thanks. When will the code changes be ready? I am not sure when I can start doing this.

"Note that I am not sure if you can easily change it, since it's a required method in the class the observables inherit from." -- I guess I would go for just write `.inverse` for each new observable class? is that going to be okay?